ocs_samples/basic_samples/OmfIngress/DotNet/readme.md

```diff
@@ -0,0 +1,173 @@
+        Name = "REPLACE_WITH_SUBSCRIPTION_NAME",
+        Description = "This is a sample OCS Data Store Subscription",
+        Type = SubscriptionType.Sds,
+        TopicId = createdTopic.Id,
+        TopicTenantId = "REPLACE_WITH_TOPIC_TENANT_ID",
+        TopicNamespaceId = "REPLACE_WITH_TOPIC_NAMESPACE_ID"
+    };
+```
+
+Then use the Ingress client to create the Subscription in OCS:
+
+```
+    Subscription createdSubscription = await omfIngressService.CreateOrUpdateSubscriptionAsync(subscription);
+```	
+
+Send OMF Messages
+-------------------
+
+OMF messages sent to OCS are translated into objects native to the Sequential Data Store. In this example, we send an OMF Type message which creates an SDS type in the data store, 
+an OMF Container message which creates an SDS stream, and then send OMF Data messages, which use the containerId in the message body to route the data to the SDS stream. 
+Refer to the data store documentation for how to view the types/streams/data in SDS. For each type of message, we first construct the message body using the OMF library:
+
+```	
+    OmfTypeMessage typeMessage = OmfMessageCreator.CreateTypeMessage(typeof(DataPointType));
+
+    OmfContainerMessage containerMessage = OmfMessageCreator.CreateContainerMessage(streamId, typeof(DataPointType));
+
+    DataPointType dataPoint = new DataPointType() { Timestamp = DateTime.UtcNow, Value = rand.NextDouble() };
+    OmfDataMessage dataMessage = OmfMessageCreator.CreateDataMessage(streamId, dataPoint);
+```
+
+Then the devices uses its own ingress client, which uses the device clientId and clientSecret to authenticate the requests. The device clientId is used to route the message
+to the Topic that the clientId is mapped to. Note that the message must be serialized before being sent.
+
+```
+    var serializedMessage = OmfMessageSerializer.Serialize(omfMessage);
+    await deviceOmfIngressService.SendOMFMessageAsync(serializedMessage);
+```
+
+Cleanup: Deleting Topics and Subscriptions
+-----------------------------------------------------
+
+In order to prevent unused resources from being left behind, this sample performs some cleanup before exiting. 
+
+Deleting Containers and Types can be achieved by constructing the same OMF messages, but instead specifying the Delete action:
+
+```
+    OmfTypeMessage typeMessage = OmfMessageCreator.CreateTypeMessage(typeof(DataPointType));
+    typeMessage.ActionType = ActionType.Delete;
+
+    OmfContainerMessage containerMessage = OmfMessageCreator.CreateContainerMessage(streamId, typeof(DataPointType));
+    containerMessage.ActionType = ActionType.Delete;  
+```
+
+Then serialize the message and send as shown in the prior section.
+
+Deleting Subscriptions and Topics can be achieved using the Ingress client and passing the corresponding object Ids:
+
+```
+    await omfIngressService.DeleteSubscriptionAsync(createdSubscription.Id);
+    await omfIngressService.DeleteTopicAsync(createdTopic.Id);
+```
```

Comment:
Do you have a test for this sample?  All samples should have an automated test with public results.    I have been bundling all of the test builds together into the same azure-pipelines.yaml at the root using public agents (I can do yaml and secrets part of this).  We can chat about this via email too.